### PR TITLE
refactor(meshratelimit): remove deprecated from field

### LINKED ISSUE #17258: MeshRateLimit: remove `from[]`
Part of #17238. Remove deprecated `from`. Update spec, validator, plugin, golden files.

### PR DESCRIPTION
**Pull request by Automaat**
## Motivation

Part of #17238. Removes the deprecated `from[]` field from MeshRateLimit.

## Implementation information

- Updated spec to remove `from` field
- Updated validator logic
- Updated plugin
- Regenerated golden files

## Supporting documentation

Closes #17258

_Generated by Sybra harness_

### CODE REVIEW

File: pkg/plugins/policies/meshratelimit/api/v1alpha1/validator.go [hunk 1]
```diff
@@ -16,13 +16,12 @@ import (
 func (r *MeshRateLimitResource) validate() error {
 	var verr validators.ValidationError
 	path := validators.RootedAt("spec")
-	if len(pointer.Deref(r.Spec.Rules)) > 0 && (len(pointer.Deref(r.Spec.To)) > 0 || len(pointer.Deref(r.Spec.From)) > 0) {
-		verr.AddViolationAt(path, "fields 'to' and 'from' must be empty when 'rules' is defined")
+	if len(pointer.Deref(r.Spec.Rules)) > 0 && len(pointer.Deref(r.Spec.To)) > 0 {
+		verr.AddViolationAt(path, "field 'to' must be empty when 'rules' is defined")
 	}
 	verr.AddErrorAt(path.Field("targetRef"), r.validateTop(r.Spec.TargetRef, inbound.AffectsInbounds(r.Spec)))
```
[the remark this answers is not in the document]
Comment (Automaat) [reply pkg/plugins/policies/meshratelimit/api/v1alpha1/validator.go:25]: Added the empty-policy guard in `pkg/plugins/policies/meshratelimit/api/v1alpha1/validator.go` so MeshRateLimit now rejects resources when both `spec.to` and `spec.rules` are absent. That landed in `b5a8467851` together with the regression coverage.

_Generated by Sybra harness_

File: pkg/plugins/policies/meshratelimit/api/v1alpha1/validator_test.go [hunk 2]
```diff
@@ -143,32 +133,12 @@ rules:
...
 			Entry("not allow invalid values", testCase{
 				inputYaml: `
 targetRef:
   kind: Dataplane
-from:
-- targetRef:
-    kind: Mesh
-  default:
+rules:
+- default:
```
[the remark this answers is not in the document]
Comment (Automaat) [reply pkg/plugins/policies/meshratelimit/api/v1alpha1/validator_test.go:141]: Added the validator regression case in `pkg/plugins/policies/meshratelimit/api/v1alpha1/validator_test.go` for the `spec.to`/`spec.rules` empty-policy case. That is included in `b5a8467851`.

_Generated by Sybra harness_

**Review by Automaat — COMMENTED** [review]
test

**Review by Automaat — COMMENTED** [review]
## Verdict: COMMENT — no blocking issues

Clean removal of the deprecated `from` field from MeshRateLimit. API type, generated files, CRDs, OpenAPI, golden fixtures, and tests are all consistent. UPGRADE.md guidance is accurate.

### Observation 1 — `configurer.go:253` — silent error swallow lacks test coverage

`ConfigureMatchedRoutesOnFilterChain` previously propagated `UnexpectedFilterConfigTypeError` when a filter chain had no HCM. It now silences that error, making non-HTTP filter chains a no-op for HTTP rate limiting. Likely correct for mixed HTTP+TCP listeners, but:
- No test exercises the skip path to document intent.
- If a filter chain should have an HCM but does not (codegen bug), the error is now invisible.

Consider a unit test that passes a non-HCM filter chain and asserts `nil` is returned.

### Observation 2 — `util.go:100` — naming clarity in `Is` method

The parameter name `err` in the new `Is(err error)` method is the *target* passed to `errors.Is`, not the wrapped error. Renaming it `target` follows Go convention and reduces reader confusion. Purely a readability nit; the logic is correct.

_Generated by Sybra harness_

File: pkg/plugins/policies/meshratelimit/plugin/xds/configurer.go [hunk 3]
```diff
@@ -250,7 +250,7 @@ func ConfigureMatchedRoutesOnFilterChain(filterChain *envoy_listener.FilterChain
 			}
 		}
 		return ConfigureMatchedRoutes(hcm.GetRouteConfig(), effectiveRules)
-	}); err != nil {
+	}); err != nil && !errors.Is(err, &listeners_v3.UnexpectedFilterConfigTypeError{}) {
```
Comment (Automaat) [inline pkg/plugins/policies/meshratelimit/plugin/xds/configurer.go:253]: Silent error swallowing is a behavioral change with no test coverage.

Previously if `UpdateHTTPConnectionManager` returned `UnexpectedFilterConfigTypeError` (no HCM on filter chain), the whole `ConfigureMatchedRoutesOnFilterChain` call failed. Now that error is swallowed and non-HTTP filter chains are silently skipped. Likely correct for mixed HTTP+TCP listeners, but no test exercises the skip path to document intent. If a filter chain should have an HCM but does not (codegen bug), the error is now invisible.

Consider adding a unit test that passes a non-HCM filter chain to `ConfigureMatchedRoutesOnFilterChain` and asserts `nil` is returned.

_Generated by Sybra harness_

File: pkg/xds/envoy/listeners/v3/util.go [hunk 4]
```diff
@@ -96,6 +96,11 @@ func (e *UnexpectedFilterConfigTypeError) Error() string {
 	return fmt.Sprintf("filter config has unexpected type: expected %T, got %T", e.expected, e.actual)
 }
 
+func (e *UnexpectedFilterConfigTypeError) Is(err error) bool {
+	_, ok := err.(*UnexpectedFilterConfigTypeError)
```
Comment (Automaat) [inline pkg/xds/envoy/listeners/v3/util.go]: The `errors.Is` parameter is conventionally named `target`, not `err` — here the parameter is the *target* checked against, not the wrapped error. Renaming clarifies intent for future readers:

```go
func (e *UnexpectedFilterConfigTypeError) Is(target error) bool {
    _, ok := target.(*UnexpectedFilterConfigTypeError)
    return ok
}
```

Purely a readability nit; the logic is correct.

_Generated by Sybra harness_

File: pkg/plugins/policies/meshratelimit/plugin/xds/configurer.go (hunk 3 above)
Comment (Automaat) [reply pkg/plugins/policies/meshratelimit/plugin/xds/configurer.go:253]: **Applied** — added a regression test in `configurer_test.go` that feeds a filter chain whose `http_connection_manager` filter carries a non-HCM config, and asserts `ConfigureMatchedRoutesOnFilterChain` returns nil (skip path) with the config left untouched (9c4790974f).

_Generated by Sybra harness_

File: pkg/xds/envoy/listeners/v3/util.go (hunk 4 above)
Comment (Automaat) [reply pkg/xds/envoy/listeners/v3/util.go]: **Applied** — renamed the `Is` parameter from `err` to `target` in `pkg/xds/envoy/listeners/v3/util.go` (9c4790974f).

_Generated by Sybra harness_

**Review by bartsmykla — APPROVED** [review]
